astropy/utils/misc.py

```diff
@@ -226,15 +226,10 @@ def find_api_page(obj, version=None, openinbrowser=True, timeout=None):
 
     # Custom request headers; see
     # https://github.com/astropy/astropy/issues/8990
-    req = urllib.request.Request(
-        baseurl + 'objects.inv', headers={'User-Agent': f'Astropy/{version}'})
-
-    if timeout is None:
-        uf = urllib.request.urlopen(req)
-    else:
-        uf = urllib.request.urlopen(req, timeout=timeout)
-
-    try:
+    url = baseurl + 'objects.inv'
+    headers={'User-Agent': f'Astropy/{version}'}
+    with get_readable_fileobj(url, encoding='binary', remote_timeout=timeout,
+            http_headers=headers) as uf:
```

Comment:
More nitpick.
```suggestion
    with get_readable_fileobj(url, encoding='binary', remote_timeout=timeout,
                              http_headers=headers) as uf:
```

Other than that, nice catch! I like this update.